python/ffsim/states/slater.py

```diff
@@ -0,0 +1,353 @@
+from __future__ import annotations
+
+import numpy as np
+import scipy.linalg
+
+
+def sample_slater(
+    rdm: np.ndarray | tuple[np.ndarray, np.ndarray],
+    chain_length: int,
+    n_chains: int,
+    n_particles_to_move: int = 1,
+    seed=None,
+) -> np.ndarray:
+    """
+    Collect samples of electronic configurations from a Slater determinant defined
+    by its one-body reduced density matrix (RDM). The sampler uses the
+    Metropolis-Hastings method.
```

Comment:
The format of a docstring should be
```python
def func():
"""Single-line summary.

Main body with extended description. Can be multiple lines.
"""
```

In the main body of the docstring, please provide enough context or background information that the `chain_length` and other arguments can be understood. I see you wrote
>chain_length: Length of each sampling chain.

but you haven't explained what a sampling chain _is_.